mapsforge-map/src/main/java/org/mapsforge/map/layer/hills/ShadingAlgorithm.java

```diff
@@ -17,8 +17,86 @@
 import org.mapsforge.core.graphics.HillshadingBitmap;
 
 import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+import java.nio.channels.FileChannel;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipInputStream;
 
 public interface ShadingAlgorithm {
+    abstract class AbsShadingAlgorithmDefaults implements ShadingAlgorithm{
```

Comment:
Please separate `ShadingAlgorithm` from its nested classes (old and new) and use a separate file for each class for simplicity / reading reasons.

Reply:
Done. That nested parent pattern is basically a crude backport of the Java 8 interface extensions (default methods etc) but I see how avoidable inheritance always comes with a readability cost.

Would raw DEM reading code deduplication implemented as a static method on RawShadingResult (or even as constructor) be OK?

Reply:
> That nested parent pattern is basically a crude backport of the Java 8 interface extensions

Then we should simplify completely the `ShadingAlgorithm`.
The library must support Java 7 for maximum compatibility.

Reply:
Yes, the transformToByteBuffer are duplicate. 

Separate class for AbsShadingAlgorithmDefaults is fine as well (convert(..) is only an internal sub-step of the public transformToByteBuffer so it would preferrable remain non-public so an abstract class is cleaner than some static helper method).

Hard-pushing an amended commit if fine for github and your preference for squashedness, right? (I'm more of a gitlab and messy unaltered commit history person, eager to see other approaches)

Reply:
> 
> 
> > That nested parent pattern is basically a crude backport of the Java 8 interface extensions
 
> The library must support Java 7 for maximum compatibility.

The pattern doesn't use Java 8 features, it tries to emulate as good as it can using nothing than Java 7

But with the top level AbsShadingAlgorithmDefaults it's conventional j7 style anyways

Reply:
Forced the manually squashed (rollback+amend) top level AbsShadingAlgorithmDefaults version.

Is github smart/aggressive enough to not pollute history with that Merge pull request commit?

Reply:
The pull request has 2 commits: a strange one and the correct with the code changes.

Reply:
Seems like my handling of upstream isn't is it should be. 

My zipped-hgt branch seems fine now, fork master required some more hand-holding (not that it matters for this PR)